### PR TITLE
🏷️ Add typings for app hang functionality

### PR DESCRIPTION
**Pull request by sandergo90**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

The sentry iOS sdk contains functionality to log a message to sentry when the app hangs.

This is not available in the types now. This pull requests adds the typings.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
Thanks @sandergo90 
Have you tested this feature specifically on RN already?

**Comment by sandergo90** [comment]
Hi! 

We have enabled it already in our application with a ts-expect-error comment ;) 

I've debugged in the iOS code, and it activated the app hanging perfectly.

File: src/js/options.ts [hunk 1]
```diff
@@ -88,16 +88,35 @@ export interface BaseReactNativeOptions {
...
+   * @default 30
+   */
   maxCacheItems?: number;
+
+  /**
+   * When enabled, the SDK tracks when the application stops responding for a specific amount of
+   * time defined by the `appHangTimeoutInterval` option.
+   *
+   * @default true
+   */
```
Comment (marandaneto) [inline src/js/options.ts:108]: I'd add a note that this is on iOS only

**Comment by marandaneto** [comment]
@sandergo90 thank you, please add a changelog entry and ready to go :)

File: src/js/options.ts (hunk 1 above)
Comment (sandergo90) [reply src/js/options.ts]: Done!

**Comment by sandergo90** [comment]
Done and done 👌

**Review by marandaneto — APPROVED** [review]

**Pull request merged**